### PR TITLE
refactored quicklinks to dataset feature toggles

### PR DESCRIPTION
**Pull request by ytqsl**
# Fixes or Implements [AB#14948](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/14948)

Task 14948: Refactor quick links

## Description

Quicklinks to use new feature toggles config

image

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/src/constants/entryType.ts [hunk 1]
```diff
@@ -97,6 +116,7 @@ entryTypeMap.set(EntryType.MedicationRequest, {
     icon: "file-medical",
     component: "MedicationRequestComponent",
     eventName: "special_authority",
+    enabled: (config) => isDatasetEnabled(config, "Medication"),
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/constants/entryType.ts:119]: This should be passing the dataset named "SpecialAuthorityRequest".  Once we've finished getting rid of the modules, we can update the enum values so these overrides won't be needed and we can just pass the EntryType as-is.

**Review by MikeLyttle — CHANGES_REQUESTED** [review]

File: Apps/WebClient/src/ClientApp/src/constants/entryType.ts (hunk 1 above)
Comment (ytqsl) [reply Apps/WebClient/src/ClientApp/src/constants/entryType.ts:119]: We have a dataset called `specialAuthority`, is this the one?

Comment (MikeLyttle) [reply Apps/WebClient/src/ClientApp/src/constants/entryType.ts]: Yes, though I think it would be better if the dataset name included request.

**Review by MikeLyttle — APPROVED** [review]

**Review by BrianMaki — COMMENTED** [review]
What about functional tests?

**Pull request merged**